oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/DurationUtils.java

```diff
@@ -216,4 +216,59 @@ private DateTime parseToDateTime(Downsampling downsampling, long time) throws Pa
 
         return dateTime;
     }
+
+    public long convertBucketTotIimestamp(boolean isStart, long bucket) throws ParseException {
```

Comment:
First, you have a typo `Timestampe`.

Then, this method should be separated into 2 methods, by removing `isStart` as a parameter. It is very unclear.
1. startBucketTime2Timestamp
1. endBucketTime2Timestamp

Reply:
This is a good suggestion

Reply:
I have the same requirement. Need to convert the TB to Timestamp to make InfluxDB RelentationPolicy work.

https://github.com/apache/skywalking/blob/484e078399f90e2b11d9bc16a83c76bbc1d4921b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java#L53-L67

Reply:
So my conversion logic remains in DurationUtils?

Reply:
Again, logic or util is not the issue. Today, the issue is the codes are not clear and separated and being injected out of ES implementation.

Reply:
I know that to optimize the elasticsearch query, you should put the code on the elasticsearch implementation class. @dmsolr will use the same conversion logic as me, why can't I put it into DurationUtils so that he can reuse this logic?

Reply:
In fact, I don't understand why we need this?  On ES, TB or TS, have any difference?

Reply:
TB represents a time bucket, such as 2020-01-17 1331 or 2020-01-17 13. TS is a time stamp such as 1579239147000. I need to convert TB to TS in order to unify time.

Reply:
I see TB means TimeBucket and its form.
I am confused about why we need to uniform them. The form(TB/TS) is not determined by the storage layer. So you covert the TB to TS on write and covert it again on the query.
In other words, whether ES depends on TS. (Maybe I miss something.)

Reply:
First, timeBucket isn't in the format you are demonstrating. Then, I am not talking who can put what codes in any place. My point is always being, put the storage optimization codes inside the storage implementation. Don't change the storage model interface unless you have to.

> why can't I put it into DurationUtils so that he can reuse this logic?

This question is really not a good way to discuss the question. I hope you wouldn't do this again. The things we discussed is not about whom, @dmsolr is showing something he is working on, his PR is in the WIP status, no review/merge.  

Also, you should understand, as an official committer, he actually really has some privileges to change APIs, it could be easier than you. Because he has been involved in source codes level a long time, and be familiar with the workflow (You could simply check his GitHub repo changelog). But, even there is a privilege, back to this case, I or he never use this priority. So, being an open source project, please don't consider discussion(arguing between you and others) in this way.